book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/entity/EventShow.java

```diff
@@ -0,0 +1,49 @@
+package org.lbcc.bms.bms_monolith.entity;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.ElementCollection;
+import jakarta.persistence.CollectionTable;
+import jakarta.persistence.Column;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.FetchType;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.lbcc.bms.bms_monolith.common.entity.BaseAuditingEntity;
+import org.lbcc.bms.bms_monolith.entity.enums.Genre;
+
+import java.time.Instant;
+import java.util.List;
+
+@Entity
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class EventShow extends BaseAuditingEntity {
+
+    @ElementCollection(targetClass = Genre.class)
+    @CollectionTable(name = "show_genre", joinColumns = @JoinColumn(name = "show_id"))
+    @Column(name = "genre")
+    @Enumerated(EnumType.STRING)
+    private List<Genre> genres;
+
+    private Instant startDate;
+    private Instant endDate;
+
+    @ManyToOne
+    @JoinColumn(name = "event_id")
+    private Event event;
+
+    @OneToMany(mappedBy = "show", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private List<SeatInShow> seatInShows;
+
+    @OneToMany(mappedBy = "show", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private List<SeatTypeInShow> seatTypeInShows;
+}
```

Comment:
Add these methods to the class

public void addSeatInShow(SeatInShow seatInShow) {
    this.seatInShows.add(seatInShow);
    seatInShow.setShow(this); // Ensures bidirectional relationship consistency
}

public void removeSeatInShow(SeatInShow seatInShow) {
    this.seatInShows.remove(seatInShow);
    seatInShow.setShow(null); // Clears the relationship
}

public void addSeatTypeInShow(SeatTypeInShow seatTypeInShow) {
    this.seatTypeInShows.add(seatTypeInShow);
    seatTypeInShow.setShow(this); // Ensures bidirectional relationship consistency
}
 @AssertTrue(message = "End date must be after start date")
    private boolean isEndDateAfterStartDate() {
        return endDate == null || startDate == null || endDate.isAfter(startDate);
    }

    public List<Genre> getGenres() {
        return Collections.unmodifiableList(genres);
    }

    public List<SeatInShow> getSeatInShows() {
        return Collections.unmodifiableList(seatInShows);
    }

    public List<SeatTypeInShow> getSeatTypeInShows() {
        return Collections.unmodifiableList(seatTypeInShows);
    }

    public void addGenre(Genre genre) {
        this.genres.add(genre);
    }

    public void removeGenre(Genre genre) {
        this.genres.remove(genre);
    }


public void removeSeatTypeInShow(SeatTypeInShow seatTypeInShow) {
    this.seatTypeInShows.remove(seatTypeInShow);
    seatTypeInShow.setShow(null); // Clears the relationship
}